v0/destinations/sendgrid/transform.js

```diff
@@ -0,0 +1,255 @@
+const logger = require("../../../logger");
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  getValueFromMessage,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  removeUndefinedAndNullValues,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getIntegrationsObj,
+  isEmptyObject
+} = require("../../util");
+const { ENDPOINT } = require("./config");
+const {
+  payloadValidator,
+  eventValidity,
+  createList,
+  createContent,
+  createAttachments,
+  constructFields
+} = require("./util");
+
+const trackResponseBuilder = async (message, { Config }) => {
+  let event = getValueFromMessage(message, "event");
+  if (!event) {
+    throw new CustomError("event is required for track call", 400);
+  }
+  event = event.trim().toLowerCase();
+  eventValidity(Config, event);
+  let payload = {};
+  const integrationsObj = getIntegrationsObj(message, "sendgrid");
+  if (!integrationsObj) {
```

Comment:
do we need to put a check for empty integrationsObj?

Reply:
no. will remove that. had kept it while testing